### PR TITLE
refactor: final cleanup - rename, remove obsolete, simplify, consolidate (SDKCF-2082)

### PR DESCRIPTION
**Pull request by imaginaris**
Sorry for long PR but this is mostly renaming.

Things done:
- IAMPreferenceBuilder add discardable
- CampaignCropType, CampaignTextAlignType, CampaignOrientation, CampaignViewType models linked
- removed unused stuff
- naming unified
- debugPrint - 'InAppMessaging' prefix

Also fixes issue with calling `requestFromServerSync` from main thread (WorkScheduler and TaskSchedulable)

### CODE REVIEW

**Comment by imaginaris** [comment]
> As discussed, the renaming/removal parts from this PR should be extracted to their own PR to make it a bit easier on the reviewers 😅

Ok! PR refreshed!

File: RInAppMessaging/Classes/Protocols/HttpRequestable.swift [hunk 1]
```diff
@@ -127,7 +127,7 @@ extension HttpRequestable {
                 let serverResponse = response as? HTTPURLResponse else {
 
                     completion(.failure(.httpError(statusCode, response, data)))
-                    CommonUtility.debugPrint("InAppMessaging: HTTP call failed.")
+                    print("InAppMessaging: HTTP call failed (\(statusCode))")
```
Comment (donnie-jp) [inline RInAppMessaging/Classes/Protocols/HttpRequestable.swift:130]: why `print` and not `debugPrint`?

File: RInAppMessaging/Classes/Services/ImpressionService.swift [hunk 2]
```diff
@@ -106,7 +106,7 @@ extension ImpressionService {
             let body = try JSONEncoder().encode(impressionRequest)
             return .success(body)
         } catch {
-            let description = "InAppMessaging: Error encoding impression request"
+            let description = "Error encoding impression request"
             CommonUtility.debugPrint("\(description): \(error)")
             reportError(description: description, data: error)
```
Comment (donnie-jp) [inline RInAppMessaging/Classes/Services/ImpressionService.swift:111]: Maybe `reportError` can include the `debugPrint` for conciseness?

File: RInAppMessaging/Classes/Protocols/HttpRequestable.swift (hunk 1 above)
Comment (imaginaris) [reply RInAppMessaging/Classes/Protocols/HttpRequestable.swift:130]: I think this type of log can be present in non-debug mode

File: RInAppMessaging/Classes/Services/ImpressionService.swift (hunk 2 above)
Comment (imaginaris) [reply RInAppMessaging/Classes/Services/ImpressionService.swift:111]: Good idea

**Pull request merged**